### PR TITLE
feat: add create-branches.sh and switch-branch.sh

### LINKED ISSUE #49: Command "header" not understood
After following the installation instructions, there are several errors when running nano:

awk.nanorc line 2: command "header" not understood
coffeescript.nanorc line 2: command "header" not understood
ini.nanorc line 2: command "header" not understood
json.nanorc line 2: command "header" not understood
makefile.nanorc line 2: command "header" not understood
nginx.nanorc line 2: command "header" not understood
perl.nanorc line 4: command "header" not understood
python.nanorc line 4: command "header" not understood
ruby.nanorc line 4: command "header" not understood
sed.nanorc line 2: command "header" not understood
sh.nanorc line 4: command "header" not understood
sh. nanorc line 11: bad regex "^\s+?[0-9A-Z_]+\+?\(\)": repetition-operator operand invalid
systemd.nanorc line 2: command "header" not understood
tcl.nanorc: line 2: command "header" not understood
yaml.nanorc line 2: command "header" not understood
zsh.nanorc line 2: bad regex "\.?zsh[^\.]+?$": repetition-operator operand invalid
zsh.nanorc line 3: command "header" not understood
zsh.nanorc line 23: bad regex "^\s+?(function\s+)?[0-9A-Z_]+\s+?\(\)": repetition-operator operand invalid

### PR DESCRIPTION
**Pull request by davidhcefx**
This is a continuation of PR https://github.com/galenguyer/nano-syntax-highlighting/pull/15

* Tested Nano versions:
  * [x] 6.2 (22.04)
  * [x] 5.4 (21.04)
  * [x] 4.8 (20.04)
  * [x] 3.2 (19.04)
  * [x] 2.9.3 (18.04)
  * [x] 2.5.3 (16.04)
  * [x] 2.2.6 (14.04)
  * [ ] <2.1.6

-------

Originally proposed in https://github.com/scopatz/nanorc/pull/370

This PR provides `create-branches.sh`, which creates the following branches:

1. **pre-6.0**: Replaced newly introduced colors, eg. `rosy`, `grey`.
1. **pre-5.0**:
  a. Replaced newly introduced colors, eg. `latte`.
  b. Replaced `light*` prefix to `bright*`.
  c. Remove `bold` and `italic` specifier.
  d. Remove `bright*` prefix from background colors.
3. **pre-4.5**: Commented out `tabgives`. Fix regex failures.
4. **pre-2.9.5**: Changed `normal` to `white` and `brightnormal` to `brightwhite`. Add quotes around `syntax` names
5. **pre-2.6.0**: Commented out `comment`.
6. **pre-2.3.3**: Commented out `linter` and `magic`.
7. **pre-2.1.5**: Commented out `header`.

Providing these branches closes https://github.com/scopatz/nanorc/issues/49 https://github.com/scopatz/nanorc/issues/104 https://github.com/scopatz/nanorc/issues/164 https://github.com/scopatz/nanorc/issues/174 https://github.com/scopatz/nanorc/issues/186 https://github.com/scopatz/nanorc/pull/189 https://github.com/scopatz/nanorc/issues/191 https://github.com/scopatz/nanorc/pull/227 https://github.com/scopatz/nanorc/issues/229 https://github.com/scopatz/nanorc/pull/236 https://github.com/scopatz/nanorc/issues/238 https://github.com/scopatz/nanorc/pull/242 https://github.com/scopatz/nanorc/issues/335 https://github.com/scopatz/nanorc/issues/349 .

~~Note that the existing `v2.9` branch is significantly out of date with `master` per https://github.com/scopatz/nanorc/compare/v2.9...master. Also, the name v2.9 doesn't really apply to more recent versions of nano.~~

The pre-2.1.5 branch works on MacOS, which has nano version 2.0.6:
```
$ /usr/bin/nano --version
 GNU nano version 2.0.6 (compiled 17:24:29, Jun  5 2020)
 Email: nano@nano-editor.org    Web: http://www.nano-editor.org/
 Compiled options: --disable-nls --enable-color --enable-extra --enable-multibuffer --enable-nanorc --enable-utf8
```
If this PR is accepted, I will submit a PR to update [readme.md](https://github.com/galenguyer/nano-syntax-highlighting/blob/master/readme.md) with a note to use these new branches. I will also update [install.sh](https://github.com/galenguyer/nano-syntax-highlighting/blob/master/install.sh) to use the logic in `switch-branch.sh` to automatically select the correct branch.

### CODE REVIEW

**Comment by davidhcefx** [comment]
After merge, just run `./tool/create-branches.sh` and enter `y` after the prompt

File: tool/switch-branch.sh [hunk 1]
```diff
@@ -1,33 +1,39 @@
...
-  NANO_BRANCH='pre-5.0'
-else
-  NANO_BRANCH='master'
-fi
+echo "Found nano version ${NANO_VER} ($(nano --version | awk '/version/ {print $4}'))"
+
+# fetch the available pre* options
+git fetch --prune origin
+
+mapfile -t branches <<<"$(git branch -l -r --format='%(refname:short)' "origin/pre-*")"
```
Comment (davidhcefx) [inline tool/switch-branch.sh]:
```
git branch -l -r
```
Better change to `git branch --list --remotes` for backward compatibility.

```
mapfile -t branches
```
Better not to use the newly introduced Bash feature **mapfile** for backward compatibility.

**Comment by davidhcefx** [comment]
Hi @galenguyer, I'm not planning to perform the test for nano version <2.1.6, so you can merge this as soon as you think all is fine.

**Comment by galenguyer** [comment]
David, as I don't currently have a computer I can easily run this on, I invited you as a contributor so you should be able to merge this and push whatever you need to. Thank you for all your contributions here, I really appreciate it!

**Comment by davidhcefx** [comment]
Thanks, I'll accept your invitation and merge this when I'm available. It's a pleasure to keep this awesome artifact ongoing as well as enhancing my everyday editor!

**Pull request merged**